### DISCUSSION TITLE
SSH configuration broken on my system?

### DISCUSSION THREAD

**Discussion by ghost** [community/community#139722] [New to GitHub]
G'day,

I'm trying to use ssh to push on Linux. It works on MacOS. I have a public and private key called ~/.ssh/github{.pub,}. I copy and paste the contents of ~/.ssh/github.pub to GitHub:

` cat ~/.ssh/github.pub | clipman store `

And I save the config. Then, I compare the sha256 checksums:

```
sha256sum ~/.ssh/github.pub
79c994cc69caa98c72bf71fda5563900c350bd1237e839dc80e4f44e22b20f61
```

On GitHub: abPMiMLinGDZ+GSibXqVA9Ouho9c9JnN6+pBqu/L3dM

`ssh -T git@github.com` returns `git@github.com: Permission denied (publickey)`. So does `git fetch`.

One thing I can think is that ssh-agent isn't running because of my system configuration (it is a broken gnome-shell that is run without gnome-session from the Linux Console because GDM keeps breaking on my PC and gnome-session exits with no debug info. There are other daemons that don't work e.g. GNOME Settings and the logout command).

Mention: New to GitHub #34904

This is a similar issue, but OP was comparing the public key and the private key. I have confirmed that I am only working with the public key.

Any ideas?

Cheers

**Comment by fruetalo182** [comment]
Hi @jackharro! You can do a couple thing,

First, ensure your SSH-Agent is running:
```
eval "$(ssh-agent -s)"
ssh-add ~/.ssh/github
```

Make sure you have a proper SSH configuration for GitHub. Add the following to ~/.ssh/config if not present:

```
Host github.com
  IdentityFile ~/.ssh/github
  AddKeysToAgent yes
  UseKeychain yes
  IdentitiesOnly yes
```

Check and set the permissions on your private key to make sure they are restricted:

```
chmod 600 ~/.ssh/github
```

Use the verbose flag to see detailed error messages:

```
ssh -vvvT git@github.com
```
